invenio_rdm_records/collections/service.py

```diff
@@ -131,3 +162,43 @@ def read_logo(self, identity, slug):
         if _exists:
             return url_for("static", filename=logo_path)
         raise LogoNotFoundError()
+
+    def read_many(self, identity, ids_=None, depth=2):
+        """Get many collections."""
+        self.require_permission(identity, "read")
+
+        ids_ = ids_ or []
+        res = self.collection_cls.resolve_many(ids_, depth=depth)
+        return CollectionList(
+            identity, res, self.collection_schema, None, self.links_item_tpl
+        )
+
+    def read_all(self, identity, depth=2):
```

Comment:
shouldn't the name be `search` to align with the usual service method naming convention?

Reply:
**Naming it search**

 I don't agree, I would expect `search` to search in the search engine. Plus `read` (and `read_many`, `read_all`) are also widely used in other services. 

Moreover, `collections_service.search` could also be ambiguous. Does it mean search collections or search records inside the collections? In my opinion the latter makes more sense, so we added `collections_service.search_records` to be explicit about it.

---

**Difference between `read_many` and `read_all`**

Read many is a specific query `SELECT  ... FROM ... WHERE  id in (1,2,3)` while read all means `SELECT ... FROM`. So I split it because:

1 - I want it to be very explicit they do different things 
2- To avoid `ids_=[]` having any implicit meaning. e.g. would it mean fetch all the collections OR fetch 0 collections?

---

**Underling `resolve` methods**

These were renamed to be consistent with the service methods.